hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java

```diff
@@ -603,4 +610,32 @@ private static void resetPriorities(RaftGroup original, RaftClient client) {
       // Not re-thrown in order to keep the main exception, if there is any.
     }
   }
+
+  /**
+   * Similar to {@link JavaUtils#attemptUntilTrue(BooleanSupplier, int, TimeDuration, String, Logger)},
+   * but:
+   * <li>takes max. {@link Duration} instead of number of attempts</li>
+   * <li>accepts {@link Duration} instead of {@link TimeDuration} for sleep time</li>
+   *
+   * @return true if attempt was successful,
+   * false if wait for condition to become true timed out or was interrupted
+   */
+  public static boolean attemptUntilTrue(BooleanSupplier condition, Duration pollInterval, Duration timeout) {
```

Comment:
Thanks @hemantk-12 for the review.

At first I considered `JavaUtils`, since it matches the class name in Ratis, but its scope in Ozone is very different (limited) compared to Ratis.  I'm not sure I'd like to grow it like that.

I chose `RatisHelper` since this new `attemptUntilTrue` is just a wrapper around the existing method in Ratis.

I'd prefer not having to move the method now.

Reply:
My understanding is the usage of `attemptUntilTrue` might increase over the time so better to have it in [HddsUtils](https://github.com/apache/ozone/blob/master/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsUtils.java) or [JavaUtils](https://github.com/apache/ozone/blob/master/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/JavaUtils.java). `RatisHelper` seems like a util/helper class for Ratis related stuff not general Ozone util.

If you think that it is better to leave it here and decide it later, I am fine with that too for now.

Reply:
We could even get rid of this wrapper if using "number of attempts" is fine.

(600 attempts with 100ms sleep seems to be equivalent to 60s max. wait)

Reply:
Using "number of attempts" is fine as well. It is equivalent to poll max time as you said.

I was thinking if we can get rid of `LambdaTestUtils#await` and `GenericTestUtils#waitFor` if we add it into JavaUtils. But it could be future improvement and can be done later.

For now, I'm OK if you leave it as it or use "number of attempts".